src/legacy/core_plugins/kibana/public/discover/embeddable/search_embeddable.ts

```diff
@@ -199,55 +227,65 @@ export class SearchEmbeddable extends Embeddable {
       }
       this.savedSearch.searchSource.getField('index').popularizeField(columnName, 1);
       columnActions.removeColumn(searchScope.columns, columnName);
-      this.customization.columns = searchScope.columns;
-      this.emitEmbeddableStateChange(this.getEmbeddableState());
+
+      this.updateInput({ columns: searchScope.columns });
     };
 
     searchScope.moveColumn = (columnName, newIndex: number) => {
       if (!searchScope.columns) {
         return;
       }
       columnActions.moveColumn(searchScope.columns, columnName, newIndex);
-      this.customization.columns = searchScope.columns;
-      this.emitEmbeddableStateChange(this.getEmbeddableState());
+      this.updateInput({ columns: searchScope.columns });
     };
 
-    searchScope.filter = (field, value, operator) => {
+    searchScope.filter = async (field, value, operator) => {
       const index = this.savedSearch.searchSource.getField('index').id;
-      const stagedFilter = {
-        field,
-        value,
-        operator,
-        index,
-      };
-      this.emitEmbeddableStateChange({
-        ...this.getEmbeddableState(),
-        stagedFilter,
+
+      let filters = this.filterManager.generate(field, value, operator, index);
+      filters = filters.map(filter => ({
+        ...filter,
+        $state: { store: 'appState' },
+      }));
+
+      await executeTriggerActions(APPLY_FILTER_TRIGGER, {
+        embeddable: this,
+        triggerContext: {
+          filters,
+        },
       });
     };
 
     this.searchScope = searchScope;
   }
 
-  private emitEmbeddableStateChange(embeddableState: EmbeddableState) {
-    this.onEmbeddableStateChanged(embeddableState);
-  }
-
-  private getEmbeddableState(): EmbeddableState {
-    return {
-      customization: this.customization,
-    };
+  public reload() {
+    this.courier.fetch();
   }
 
   private pushContainerStateParamsToScope(searchScope: SearchScope) {
     // If there is column or sort data on the panel, that means the original columns or sort settings have
     // been overridden in a dashboard.
-
-    searchScope.columns = this.customization.columns || this.savedSearch.columns;
-    searchScope.sort = this.customization.sort || this.savedSearch.sort;
+    searchScope.columns = this.input.columns || this.savedSearch.columns;
+    searchScope.sort = this.input.sort || this.savedSearch.sort;
     searchScope.sharedItemTitle = this.panelTitle;
 
-    this.filtersSearchSource.setField('filter', this.filters);
-    this.filtersSearchSource.setField('query', this.query);
+    // Awful hack to get search sources to send out an initial query. Angular should be going away
+    // soon and we can get rid of this.
+    if (searchScope.searchSource) {
```

Comment:
lets try to find a different solution for this

Reply:
Done! I think all I needed was the `courier.fetch` call at the bottom and this became unnecessary when that was added.